sys/hashes/sha1.c

```diff
@@ -183,7 +183,7 @@ void sha1_init_hmac(sha1_context *s, const unsigned char *key, size_t key_length
         while(key_length--) {
             sha1_update_byte(s, (unsigned char) *key++);
         }
-        memcpy(s->key_buffer, sha1_final(s), SHA1_DIGEST_LENGTH);
+        sha1_final(s, s->key_buffer);
```

Comment:
I didn't change all the functions in the headers but stuck to the three hash-calcualting functions, leaving the hmac parts alone.
But you're rigth, this should probably be done in one go, too.